src/current/v23.1/changefeed-messages.md

```diff
@@ -23,35 +23,103 @@ This page describes the format and behavior of changefeed messages. You will fin
+        - `updated`, which contains the [updated](create-changefeed.html#updated-option) timestamp.
+        - `resolved`, which is emitted for records representing [resolved](#resolved-messages) timestamps. These records do not include an `after` value since they only function as checkpoints.
     - For [`INSERT`](insert.html) and [`UPDATE`](update.html), the current state of the row inserted or updated.
     - For [`DELETE`](delete.html), `null`.
 
-For example:
+{{site.data.alerts.callout_info}}
+If you use the `envelope` option to alter the changefeed message fields, your messages may not contain one or more of the values noted in the preceding list. As an example, when emitting to a Kafka sink, you can limit messages to just the changed key value by using the `envelope` option set to [`key_only`](#key_only). For more detail, refer to [Message envelopes](#message-envelopes).
+{{site.data.alerts.end}}
+
+For example, changefeeds emitting to a sink will have the default message format:
 
 Statement                                      | Response
 -----------------------------------------------+-----------------------------------------------------------------------
 `INSERT INTO office_dogs VALUES (1, 'Petee');` | JSON: `[1]	{"after": {"id": 1, "name": "Petee"}}` </br>Avro: `{"id":{"long":1}}	{"after":{"office_dogs":{"id":{"long":1},"name":{"string":"Petee"}}}}`
 `DELETE FROM office_dogs WHERE name = 'Petee'` | JSON: `[1]	{"after": null}` </br>Avro: `{"id":{"long":1}}	{"after":null}`
 
-To limit messages to just the changed key value, use the [`envelope`](create-changefeed.html#options) option set to `key_only`.
+When a changefeed targets a table with multiple column families, the family name is appended to the table name as part of the topic. Refer to [Tables with columns families in changefeeds](changefeeds-on-tables-with-column-families.html#message-format) for guidance.
 
-When a changefeed targets a table with multiple column families, the family name is appended to the table name as part of the topic. See [Tables with columns families in changefeeds](changefeeds-on-tables-with-column-families.html#message-format) for guidance.
-
-For webhook sinks, the response format arrives as a batch of changefeed messages with a `payload` and `length`. Batching is done with a per-key guarantee, which means that messages with the same key are considered for the same batch. Note that batches are only collected for row updates and not [resolved timestamps](create-changefeed.html#resolved-option):
+For [webhook sinks](changefeed-sinks.html#webhook-sink), the response format arrives as a batch of changefeed messages with a `payload` and `length`. Batching is done with a per-key guarantee, which means that messages with the same key are considered for the same batch. Note that batches are only collected for row updates and not [resolved timestamps](create-changefeed.html#resolved-option):
```

Comment:
Sorry, not part of this PR, but I'm noticing that I don't understand what "messages with the same key are considered for the same batch" means.

Reply:
Hmmm, re-reading not sure what I meant by this when I introduced webhooks into the docs. I assume I was trying to get at how batches are evaluated? Do you know how that would involve the key?

Reply:
Okay, I think I get it. Webhook sinks obey the same per-key ordering guarantee as other sinks, so they can't make multiple requests in parallel if they contain different events concerning the same primary key--then the order the server received them would be nondeterministic. This is true whether or not the messages are being batched--if you have batching turned off, you might get a message about k1 and k2 at the same time, but you won't get another message about k1 until you've acknowledged the first one. If the messages are being batched, you won't get two batches at the same time with overlapping keys, but you might get a single batch containing two messages about k1, since you can tell the order from their position within the batch.

Reply:
I have added more detail based on your feedback. I think I have captured a better few sentences here. I do have an issue coming up to improve the ordering guarantee + schema change/backfill parts of this message page, so I hope to make this kind of info even clearer.